### ISSUE TITLE
FlutterLocation.java:280: error: lambda expressions are not supported in -source 7

### ISSUE THREAD

**Issue by accurithm** [Lyokone/flutterlocation#576]
**Describe the bug**
Getting this error when trying to build for location 4.2.2

.pub-cache/hosted/pub.dartlang.org/location-4.2.0/android/src/main/java/com/lyokone/location/FlutterLocation.java:280: error: lambda expressions are not supported in -source 7
            mMessageListener = (message, timestamp) -> {
                                                    ^
  (use -source 8 or higher to enable lambda expressions)
1 error

**Expected behavior**
Should build, as the project structure has the API level 11 support which includes lambda

**Tested on:**
N/A Build error

**Other plugins:**
 - List of others Flutter plugins that could interfere

**Additional logs**

``` shell
some logs
```

**Comment by accurithm** [comment]
I was able to temporarily solve this problem by adding the following to location-4.2.0/android/build.gradle

android {
    compileSdkVersion 30

    sourceSets {
        main.java.srcDirs += 'src/main/kotlin'
    }
    defaultConfig {
        minSdkVersion 16
        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
    }

	**compileOptions {
		targetCompatibility JavaVersion.VERSION_1_8
		sourceCompatibility JavaVersion.VERSION_1_8
	}**

}

**Comment by EarlPomeroy** [comment]
Same issue in 4.2.0. An alternate solution is to explicitly use version 4.1.1 in your pubspec.yaml.

**Comment by hayatbangash55** [comment]
getting the same error and not able to resolve it by adding these lines in gradle

compileOptions {
targetCompatibility JavaVersion.VERSION_1_8
sourceCompatibility JavaVersion.VERSION_1_8
}

**Comment by Lyokone** [comment]
Hi, 
I have no issue running the example app with the `4.2.0` version on a `flutter clean` folder.
Can you [check](https://github.com/Lyokone/flutterlocation/tree/master/packages/location/example/android) or share your app build.gradle ?

**Comment by Lyokone** [comment]
I believe you can use Android Studio to migrate your app to Java 8, by opening the `android` folder.

**Comment by hayatbangash55** [comment]
buildscript {
    ext.kotlin_version = '1.3.50'
    repositories {
        google()
        jcenter()
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:4.1.0'
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
    }
}

allprojects {
    repositories {
        google()
        jcenter()
    }
}

rootProject.buildDir = '../build'
subprojects {
    project.buildDir = "${rootProject.buildDir}/${project.name}"
}
subprojects {
    project.evaluationDependsOn(':app')
}

task clean(type: Delete) {
    delete rootProject.buildDir
}

**Comment by Lyokone** [comment]
Does it work by update the content of your `build.gradle` with [this](https://github.com/Lyokone/flutterlocation/blob/master/packages/location/example/android/build.gradle)
and your `android/gradle/wrapper/gradle-wrapper.properties`with 
```
distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-all.zip
``` 
? 

Don't forget to run `flutter clean`

**Comment by hayatbangash55** [comment]
> Does it work by update the content of your `build.gradle` with [this](https://github.com/Lyokone/flutterlocation/blob/master/packages/location/example/android/build.gradle)
> and your `android/gradle/wrapper/gradle-wrapper.properties`with
> 
> ```
> distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-all.zip
> ```
> 
> ?
> 
> Don't forget to run `flutter clean`

Yes it worked Thanks.

**Comment by zingchar** [comment]
> Same issue in 4.2.0. An alternate solution is to explicitly use version 4.1.1 in your pubspec.yaml.

I made a new flutter project. added location: ^4.1.1. still it throws an error --

`/Users/test/Work/install/flutter/.pub-cache/hosted/pub.dartlang.org/location-4.2.0/android/src/main/java/com/lyokone/location/FlutterLocation.java:280: error: lambda expressions are not supported in -source 1.7
            mMessageListener = (message, timestamp) -> {
                                                    ^
  (use -source 8 or higher to enable lambda expressions)
1 error
`

even though I have only added 4.1.1 wonder why its showing 4.2.0 in the error

**Comment by Yash-Garg** [comment]
Facing same issue, have tried all the suggested methods given above!

**Comment by LostInDarkMath** [comment]
Same issue. Looking forward for a fix :)

**Comment by reynaldi18** [comment]
Same issue :(

**Comment by sammy44nts** [comment]
Fixed by ugrading **Gradle** & **gradle-wrapper** versions:
``` diff
index 3d46d1f..9f48700 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -1,16 +1,16 @@
 buildscript {
     ext {
-        kotlin_version = '1.4.21'
+        kotlin_version = '1.5.10'
     }
     repositories {
         google()
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.1.2'
+        classpath 'com.android.tools.build:gradle:4.2.1'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
     }
 }
 
index 66e249a..833a9a4 100644
--- a/android/gradle/wrapper/gradle-wrapper.properties
+++ b/android/gradle/wrapper/gradle-wrapper.properties
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-all.zip
```

**Comment by AliYar-Khan** [comment]
> 
> 
> > Same issue in 4.2.0. An alternate solution is to explicitly use version 4.1.1 in your pubspec.yaml.
> 
> I made a new flutter project. added location: ^4.1.1. still it throws an error --
> 
> `/Users/test/Work/install/flutter/.pub-cache/hosted/pub.dartlang.org/location-4.2.0/android/src/main/java/com/lyokone/location/FlutterLocation.java:280: error: lambda expressions are not supported in -source 1.7 mMessageListener = (message, timestamp) -> { ^ (use -source 8 or higher to enable lambda expressions) 1 error `
> 
> even though I have only added 4.1.1 wonder why its showing 4.2.0 in the error
> 
> UPDATE --
> I set the package version in pubspec.yml as '4.1.1'. deleted the .pub-cache folder along with pubspec.yml.lock file. did flutter clean and flutter pub get. now its locked to 4.1.1 and works for the time being.

where is .pub-cache folder located ?

**Comment by sammy44nts** [comment]
> > > Same issue in 4.2.0. An alternate solution is to explicitly use version 4.1.1 in your pubspec.yaml.
> > 
> > 
> > I made a new flutter project. added location: ^4.1.1. still it throws an error --
> > `/Users/test/Work/install/flutter/.pub-cache/hosted/pub.dartlang.org/location-4.2.0/android/src/main/java/com/lyokone/location/FlutterLocation.java:280: error: lambda expressions are not supported in -source 1.7 mMessageListener = (message, timestamp) -> { ^ (use -source 8 or higher to enable lambda expressions) 1 error `
> > even though I have only added 4.1.1 wonder why its showing 4.2.0 in the error
> > UPDATE --
> > I set the package version in pubspec.yml as '4.1.1'. deleted the .pub-cache folder along with pubspec.yml.lock file. did flutter clean and flutter pub get. now its locked to 4.1.1 and works for the time being.
> 
> where is .pub-cache folder located ?

Your root folder but it is better to simply perform ` flutter clean` in command line.

**Comment by Yash-Garg** [comment]
> Fixed by ugrading **Gradle** & **gradle-wrapper** versions:
> 
> ```diff
> index 3d46d1f..9f48700 100644
> --- a/android/build.gradle
> +++ b/android/build.gradle
> @@ -1,16 +1,16 @@
>  buildscript {
>      ext {
> -        kotlin_version = '1.4.21'
> +        kotlin_version = '1.5.10'
>      }
>      repositories {
>          google()
>          jcenter()
>      }
>      dependencies {
> -        classpath 'com.android.tools.build:gradle:4.1.2'
> +        classpath 'com.android.tools.build:gradle:4.2.1'
>          classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
>      }
>  }
>  
> index 66e249a..833a9a4 100644
> --- a/android/gradle/wrapper/gradle-wrapper.properties
> +++ b/android/gradle/wrapper/gradle-wrapper.properties
> @@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
>  distributionPath=wrapper/dists
>  zipStoreBase=GRADLE_USER_HOME
>  zipStorePath=wrapper/dists
> -distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-all.zip
> +distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-all.zip
> ```

This worked perfectly fine for me!

**Comment by AliYar-Khan** [comment]
> 
> 
> > Fixed by ugrading **Gradle** & **gradle-wrapper** versions:
> > ```diff
> > index 3d46d1f..9f48700 100644
> > --- a/android/build.gradle
> > +++ b/android/build.gradle
> > @@ -1,16 +1,16 @@
> >  buildscript {
> >      ext {
> > -        kotlin_version = '1.4.21'
> > +        kotlin_version = '1.5.10'
> >      }
> >      repositories {
> >          google()
> >          jcenter()
> >      }
> >      dependencies {
> > -        classpath 'com.android.tools.build:gradle:4.1.2'
> > +        classpath 'com.android.tools.build:gradle:4.2.1'
> >          classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
> >      }
> >  }
> >  
> > index 66e249a..833a9a4 100644
> > --- a/android/gradle/wrapper/gradle-wrapper.properties
> > +++ b/android/gradle/wrapper/gradle-wrapper.properties
> > @@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
> >  distributionPath=wrapper/dists
> >  zipStoreBase=GRADLE_USER_HOME
> >  zipStorePath=wrapper/dists
> > -distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-all.zip
> > +distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-all.zip
> > ```
> 
> This worked perfectly fine for me!

I have to do this for my project or plugin ... Also tell me about you flutter version and dart version ?

**Comment by zenovicharis** [comment]
> **compileOptions** {
> **targetCompatibility** JavaVersion.VERSION_1_8
> **sourceCompatibility** JavaVersion.VERSION_1_8
> **}**

Works for me, it is good enough to continue until resolve comes

**Comment by Lyokone** [comment]
> I have to do this for my project or plugin ... Also tell me about you flutter version and dart version ?
You can do that just for your project.

**Comment by Lyokone** [comment]
I have released `4.2.1` that should force Java 8 on the package and prevent this error. Let me know if it's okay for you :)

**Comment by omtodkar** [comment]
> I have to do this for my project or plugin ... Also tell me about you flutter version and dart version ?

@AliYar-Khan it worked for me with flutter version `2.2.0` and dart `2.13.1`

**Comment by StephanCassel** [comment]
Same here. 
- Delete pubspec.lock
- flutter clean
- flutter pub get 
> back in business :-)

**Comment by StephanCassel** [comment]
Thanks! 

4.2.1 works fine

• Flutter version 2.2.1 
• Dart version 2.13.1

**Comment by hayatbangash55** [comment]
> I have released `4.2.1` that should force Java 8 on the package and prevent this error. Let me know if it's okay for you :)

Thanks alot. Working fine and reverted back my gradle settings.

**Comment by LostInDarkMath** [comment]
> 
> 
> I have released `4.2.1` that should force Java 8 on the package and prevent this error. Let me know if it's okay for you :)

Hi @Lyokone Does the update break support to Android API level 18? I think this incompatible. On my Android 4.4.2 phone, `location 4.1.1.` works fine, but `4.2.3` crashes.

```
06-13 17:47:39.144 19280-19280/com.conngenial.conngenial W/AccessibilityBridge: can't invoke getChildId with reflection
06-13 17:47:39.434 19280-19280/com.conngenial.conngenial I/dalvikvm: Failed resolving Lcom/lyokone/location/-$$Lambda$FlutterLocation$f8XcZ8MqI0jurqryNosbwR0gzsU; interface 285 'Landroid/location/OnNmeaMessageListener;'
06-13 17:47:39.434 19280-19280/com.conngenial.conngenial W/dalvikvm: Link of class 'Lcom/lyokone/location/-$$Lambda$FlutterLocation$f8XcZ8MqI0jurqryNosbwR0gzsU;' failed
06-13 17:47:39.434 19280-19280/com.conngenial.conngenial E/dalvikvm: Could not find class 'com.lyokone.location.-$$Lambda$FlutterLocation$f8XcZ8MqI0jurqryNosbwR0gzsU', referenced from method com.lyokone.location.FlutterLocation.createLocationCallback
06-13 17:47:39.434 19280-19280/com.conngenial.conngenial W/dalvikvm: VFY: unable to resolve new-instance 6876 (Lcom/lyokone/location/-$$Lambda$FlutterLocation$f8XcZ8MqI0jurqryNosbwR0gzsU;) in Lcom/lyokone/location/FlutterLocation;
06-13 17:47:39.434 19280-19280/com.conngenial.conngenial I/dalvikvm: Could not find method android.location.LocationManager.isLocationEnabled, referenced from method com.lyokone.location.FlutterLocation.checkServiceEnabled
06-13 17:47:39.434 19280-19280/com.conngenial.conngenial W/dalvikvm: VFY: unable to resolve virtual method 1633: Landroid/location/LocationManager;.isLocationEnabled ()Z
06-13 17:47:39.434 19280-19280/com.conngenial.conngenial I/dalvikvm: Could not find method android.location.LocationManager.addNmeaListener, referenced from method com.lyokone.location.FlutterLocation.lambda$startRequestingLocation$2$FlutterLocation
```

Should I open a new issue for this?
